include/xtypes/DynamicData.hpp

```diff
@@ -447,6 +615,131 @@ class WritableDynamicDataRef : public ReadableDynamicDataRef
         catch(bool value) { return value; }
     }
 
+    /// \brief Class used for iterate WritableDynamicDataRef
+    class Iterator : public ReadableDynamicDataRef::Iterator
+    {
+    public:
+        Iterator(
+                WritableDynamicDataRef& ref)
+            : ReadableDynamicDataRef::Iterator(ref, false)
+        {}
+
+        WritableDynamicDataRef operator * ()
+        {
+            return ReadableDynamicDataRef::Iterator::operator*();
+        }
+
+    protected:
+
+        friend class WritableDynamicDataRef;
+
+        Iterator(
+                WritableDynamicDataRef& ref,
+                bool end)
+            : ReadableDynamicDataRef::Iterator(ref, end)
+        {
+        }
+    };
+
+    /// \brief Returns the initial iterator of a collection dynamic data.
+    /// \pre The DynamicData must represent a CollectionType.
+    /// \returns The initial iterator.
+    Iterator begin()
+    {
+        assert(type_.is_collection_type());
+        return Iterator(*this);
+    }
+
+    /// \brief Returns the final iterator of a collection dynamic data.
+    /// \pre The DynamicData must represent a CollectionType.
+    /// \returns The final iterator.
+    Iterator end()
+    {
+        assert(type_.is_collection_type());
+        return Iterator(*this, true);
+    }
+
+    class MemberPair : public ReadableDynamicDataRef::MemberPair
+    {
+    public:
+        MemberPair(
+                const Member& member,
+                uint8_t* data)
+            : ReadableDynamicDataRef::MemberPair(member, data)
+        {}
+
+        WritableDynamicDataRef data()
+        {
+            return WritableDynamicDataRef(member_.type(), data_);
+        }
+
+        ReadableDynamicDataRef data() const
+        {
+            return ReadableDynamicDataRef::MemberPair::data();
+        }
+
+    };
+
+    class MemberIterator : public Iterator
+    {
+    public:
+
+        MemberIterator(
+                WritableDynamicDataRef& ref)
+            : Iterator(ref, false)
+            , ref_(ref)
+        {}
+
+        MemberPair operator * ()
+        {
+            const AggregationType& aggregation = static_cast<const AggregationType&>(type_);
+            return MemberPair(
+                aggregation.member(index_),
+                data_ + aggregation.member(index_).offset());
+        }
+
+        MemberIterator begin()
+        {
```

Comment:
Could you call the readable version that already checks this (see similar ocurrences)